RHTAP-5179: Splitting TPA and Chicken Realm Charts

otaviof opened the pull request:
Making sure the Keycloak Chicken realm is imported before TPA takes place.

otaviof committed:
chore: Rename Realm Templates

Into new chart `tssc-tpa-realm`.

otaviof committed:
chore: Move Script and Documentation

Hack script and contributing documentation from `tssc-tpa` moved into
`tssc-tpa-realm`. These two are dealing with Keycloak Realm things.

otaviof committed:
fix: Remove TPA Realm Management from `tssc-tpa`

otaviof committed:
fix: Dedicated TPA Realm Management Chart

otaviof commented:
/test all

Roming22 commented on installer/charts/tssc-tpa-realm/templates/_copy-scripts.tpl:
This should be a symlink to `../../_common/_copy-scripts.tpl`.

otaviof replied:
Indeed, thanks for the heads up. Just fixed on the latest push.

Roming22 commented:
/retest

Roming22 commented on installer/charts/values.yaml.tpl:
```diff
@@ -309,22 +309,40 @@ developerHub:
   {{- $protocol = "http" }}
 {{- end }}
+{{- $tpaRealmPath := "realms/chicken" }}
+{{- $tpaOIDCIssuerURL := printf "%s://%s/%s" $protocol $keycloakRouteHost $tpaRealmPath -}}
```
`-}}` -> `}}`.

otaviof replied:
Thanks, also fixed.

Roming22 reviewed (changes requested):
I came across this issue in my test.

Roming22 commented on installer/charts/tssc-tpa-realm/templates/_service-account.tpl:
You are missing the service-account.yaml that goes with this file.

otaviof replied:
Indeed, this file is missing, let me amend this 🏃

otaviof committed:
chore: Helm Pre-Install Hook to Wait for SSO Endpoint

Making sure the `tssc-tpa` Helm chart waits for the SSO endpoint before
installation and upgrades.

dperaza4dustbit commented on installer/charts/values.yaml.tpl:
```diff
@@ -309,22 +309,40 @@ developerHub:
   {{- $protocol = "http" }}
 {{- end }}
+{{- $tpaRealmPath := "realms/chicken" }}
+{{- $tpaOIDCIssuerURL := printf "%s://%s/%s" $protocol $keycloakRouteHost $tpaRealmPath }}
```
why not test for full url which is what makes TPA server failed: https://redhat-internal.slack.com/archives/C065GQMTLN4/p1747788950386529?thread_ts=1747754309.792459&cid=C065GQMTLN4

dperaza4dustbit reviewed (commented):
Looks good just a comment on the oidc url being tested
